src/pretix/api/views/item.py

```diff
@@ -28,18 +29,149 @@ class Meta:
+            data={**self.request.data, **{'ORDER': serializer.instance.position}, **{'id': serializer.instance.pk}}
+        )
+
+    def perform_update(self, serializer):
+        variation = get_object_or_404(ItemVariation, pk=self.kwargs['pk'])
+        serializer.save(variation=variation)
+        variation.item.log_action(
+            'pretix.event.item.variation.changed',
+            user=self.request.user,
+            api_token=(self.request.auth if isinstance(self.request.auth, TeamAPIToken) else None),
+            data={**self.request.data['value'], **{'id': self.kwargs['pk']}}
+        )
+
+    def destroy(self, request, *args, **kwargs):
+        variation = get_object_or_404(ItemVariation, pk=self.kwargs['pk'])
+
+        ItemVariation.clean_order_positions(variation)
+        ItemVariation.clean_cart_positions(variation)
+
+        super().perform_destroy(variation)
+        variation.item.log_action(
+            'pretix.event.item.variation.deleted',
+            user=self.request.user,
+            api_token=(self.request.auth if isinstance(self.request.auth, TeamAPIToken) else None),
+            data={
+                'value': variation.value,
+                'id': self.kwargs['pk']
+            }
+        )
+
+
+class ItemAddOnViewSet(viewsets.ModelViewSet):
+    serializer_class = ItemAddOnSerializer
+    queryset = ItemAddOn.objects.none()
+    filter_backends = (DjangoFilterBackend, OrderingFilter,)
+    ordering_fields = ('id', 'position')
+    ordering = ('id',)
+    permission = 'can_change_items'
+    write_permission = 'can_change_items'
+
+    def get_queryset(self):
+        item = get_object_or_404(Item, pk=self.kwargs['item'])
+        return item.addons.all()
+
+    def perform_create(self, serializer):
+        item = get_object_or_404(Item, pk=self.kwargs['item'])
+        category = get_object_or_404(ItemCategory, pk=self.request.data['addon_category'])
+        serializer.save(item=item, addon_category=category)
+        item.log_action(
+            'pretix.event.item.addons.added',
+            user=self.request.user,
+            api_token=(self.request.auth if isinstance(self.request.auth, TeamAPIToken) else None),
+            data={**self.request.data, **{'ORDER': serializer.instance.position}, **{'id': serializer.instance.pk}}
+        )
+
+    def perform_update(self, serializer):
+        addon = get_object_or_404(ItemAddOn, pk=self.kwargs['pk'])
+        serializer.save(addon=addon)
+        addon.item.log_action(
+            'pretix.event.item.addons.changed',
+            user=self.request.user,
+            api_token=(self.request.auth if isinstance(self.request.auth, TeamAPIToken) else None),
+            data={**self.request.data['value'], **{'id': self.kwargs['pk']}}
+        )
+
+    def perform_destroy(self, serializer):
+        addon = get_object_or_404(ItemAddOn, pk=self.kwargs['pk'])
```

Comment:
the signature of this is `` def perform_destroy(self, instance):``, not ``serializer``, you don't need to retrieve the addon yourself